crates/ibc/src/applications/transfer/error.rs

```diff
@@ -15,134 +15,104 @@ use crate::core::ics24_host::identifier::{ChannelId, PortId};
-            |_| { "base denomination is empty" },
-
-        InvalidTracePortId
-            { pos: usize }
-            [ ValidationError ]
-            | e | { format_args!("invalid port id in trace at position: {0}", e.pos) },
-
-        InvalidTraceChannelId
-            { pos: usize }
-            [ ValidationError ]
-            | e | { format_args!("invalid channel id in trace at position: {0}", e.pos) },
-
-        InvalidTraceLength
-            { len: usize }
-            | e | { format_args!("trace length must be even but got: {0}", e.len) },
-
-        InvalidAmount
-            [ TraceError<FromDecStrErr> ]
-            | _ | { "invalid amount" },
-
-        InvalidToken
-            | _ | { "invalid token" },
-
-        Signer
-            [ SignerError ]
-            | _ | { "failed to parse signer" },
-
-        MissingDenomIbcPrefix
-            | _ | { "missing 'ibc/' prefix in denomination" },
-
-        MalformedHashDenom
-            | _ | { "hashed denom must be of the form 'ibc/{Hash}'" },
-
-        ParseHex
-            [ TraceError<EncodingError> ]
-            | _ | { "invalid hex string" },
-
-        ChannelNotUnordered
-            { order: Order }
-            | e | { format_args!("expected '{0}' channel, got '{1}'", Order::Unordered, e.order) },
-
-        InvalidVersion
-            { version: Version }
-            | e | { format_args!("expected version '{0}', got '{1}'", Version::ics20(), e.version) },
-
-        InvalidCounterpartyVersion
-            { version: Version }
-            | e | { format_args!("expected counterparty version '{0}', got '{1}'", Version::ics20(), e.version) },
-
-        CantCloseChannel
-            | _ | { "channel cannot be closed" },
-
-        PacketDataDeserialization
-            | _ | { "failed to deserialize packet data" },
-
-        AckDeserialization
-            | _ | { "failed to deserialize acknowledgement" },
-
-        ReceiveDisabled
-            | _ | { "receive is not enabled" },
-
-        SendDisabled
-            | _ | { "send is not enabled" },
-
-        ParseAccountFailure
-            | _ | { "failed to parse as AccountId" },
-
-        InvalidPort
-            { port_id: PortId, exp_port_id: PortId }
-            | e | { format_args!("invalid port: '{0}', expected '{1}'", e.port_id, e.exp_port_id) },
-
-        TraceNotFound
-            | _ | { "no trace associated with specified hash" },
-
-        DecodeRawMsg
-            [ TraceError<TendermintProtoError> ]
-            | _ | { "error decoding raw msg" },
-
-        UnknownMsgType
-            { msg_type: String }
-            | e | { format_args!("unknown msg type: {0}", e.msg_type) },
-
-        InvalidCoin
-            { coin: String }
-            | e | { format_args!("invalid coin string: {}", e.coin) },
-
-        Utf8Decode
-            [ TraceError<Utf8Error> ]
-            | _ | { "error decoding raw bytes as UTF8 string" },
-    }
+#[cfg(feature = "std")]
+impl std::error::Error for Error {}
```

Comment:
I think it would be nice to implement this. Once `core::error::Error` is stabilized we can rename `std::error::Error` to `core::error::Error` while the implementation remains the same -> 
```rust
#[cfg(feature = "std")]
impl std::error::Error for Error {
    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
        match self {
            Error::Ics04Channel(e) => Some(e),
            _ => None // TODO
        }
    }
}
```